python/src/aiconfig/default_parsers/openai_vision.py

```diff
@@ -0,0 +1,427 @@
+
+                content = message["content"]
+                input = {}
+                # Message can contain text and multiple images
+                for val in content:
+                    if val["type"] == "text":
+                        input["data"] = val["text"]
+                    if val["type"] == "image_url":
+                        if input.get("attachments") is None:
+                            input["attachments"] = []
+                        image_url: str = val["image_url"]["url"]
+                        attachment_kind = (
+                            "base64"
+                            if image_url.startswith("data:")
+                            else "file_uri"
+                        )
+                        attachment_data = AttachmentDataWithStringValue(
+                            kind=attachment_kind, value=image_url
+                        )
+                        input["attachments"].append(
+                            Attachment(
+                                data=attachment_data, mime_type="image/*"
+                            )
+                        )
+
+                # openai sdk currently only supports images with user messages
+                assistant_output = []
+                if assistant_response is not None:
+                    output_data = build_output_data(assistant_response)
+                    metadata = {
+                        "raw_response": assistant_response,
+                        role: "assistant",
+                    }
+                    assistant_output = [
+                        ExecuteResult(
+                            output_type="execute_result",
+                            execution_count=None,
+                            data=output_data,
+                            metadata=metadata,
+                        )
+                    ]
+                prompt = Prompt(
+                    name=new_prompt_name,
+                    input=PromptInput(
+                        data=input.get("data"),
+                        attachments=input.get("attachments"),
+                    ),
+                    metadata=PromptMetadata(
+                        model=copy.deepcopy(model_metadata),
+                        parameters=parameters,
+                        remember_chat_context=True,
+                    ),
+                    outputs=assistant_output,
+                )
+                prompts.append(prompt)
+            elif i == 0 and role == "assistant":
+                # If the first message is an assistant message,
+                # build a prompt with an empty input,
+                # and the assistant response as the output
+
+                # Pull assistant response
+                assistant_output = build_output_data(
+                    conversation_data["messages"][i]
+                )
+                prompt = Prompt(
+                    name=f"{prompt_name}_{len(prompts) + 1}",
```

Comment:
Why `len(prompts) + 1` instead of `i`?

Reply:
Answered above. But basically len(prompts) accounts for user message count (one prompt per user message) whereas i would include assistant message (output) count